go/vt/sqlparser/token.go

```diff
@@ -1063,6 +1066,7 @@ func (tkn *Tokenizer) scanMySQLSpecificComment() (int, []byte) {
 	}
 	_, sql := ExtractMysqlComment(buffer.String())
 	tkn.specialComment = NewStringTokenizer(sql)
+	tkn.specialPosOffset = strings.Index(string(tkn.buf), sql)
```

Comment:
Pretty sure this doesn't work in the general case:

1) tkn.next() can change the content of tkn.buf
2) tkn.buf can be smaller than the sql returned by the (unbounded) buffer returned by the comment extraction process

Need to figure something else out.

You should be able to test this by playing with defaultBufSize (or otherwise making it smaller) in tests.